src/NumericInput.js

```diff
@@ -49,6 +49,12 @@ export default class NumericInput extends PureComponent {
         this.props.onChange(e);
     }
 
+    componentWillUpdate(nProps, nState){
+        if (this.state.value.length - nState.value.length === 2) {
```

Comment:
@qingweibinary 2 because our problem begin immediately after the last decimal place. As soon as the very last decimal place is deleted, Our value is considered to be a whole number. Thats is `1.` is rendered as `1`  by react. So we are interested on the decimal number before the decimal itself. Therefore the last number and the decimal itself adds the length of 2 to the state length.  `0.` would be rendered as `0` nonetheless.  We are appending zero because The user may decide to clear all the decimal whole numbers before the decimal places. that is like deleting .1 from `.1` leaving our barrier value empty.  This however is optional considering we are using a controlled input type and our barrierVal has a default value. So we may choose to remove or clear the 0 string.